simulations/CC_cycle.py

```diff
@@ -91,14 +93,23 @@ def terminate_check(t, SV, SVdot, return_val, inputs):
         i_data = currents[i]*np.ones_like(solution.values.t)
         cycle_number = int(i+1-equil)*np.ones_like(solution.values.t)
         cycle_capacity = 0.1*solution.values.t*abs(i_data)/3600
+        voltage = solution.values.y.T[-7]
+        power_density = voltage*i_data
+        energy_density = [0]
+        for item, value in enumerate(cycle_capacity):
+            if item == 0:
+                continue
+            else:
+                delta_capacity = value - cycle_capacity[item-1]
+                energy_density.append(voltage[item]*delta_capacity +energy_density[item-1])
 
         # Append the current data array to any preexisting data, for output.
         # If this is the first step, create the output data array.
         if i: # Not the first step. 'data_out' already exists:
             # Stack the times, the current at each time step, and the solution
             # vector at each time step into a single data array.
             SV = np.vstack((solution.values.t+data_out[0,-1], cycle_number,
-                i_data, cycle_capacity, solution.values.y.T))
+                i_data, cycle_capacity, power_density, energy_density, solution.values.y.T))
```

Comment:
See above - there is just one power density and one energy density, really, per simulation.  Do we store them separately?

Another idea is we just calculate the cumulative power and energy densities, as a function of time, and the final value in this array is the single number that I calculate above.  Then we can store it with the solution, and the "final number" is easily extractable as the last element in the array...

Reply:
Note, if we do save the cumulative values, I think the power density still  requires some editing from what you currently have.  But it likely would involve a for loop, so my current approach doesn't work either.  More like a blend of our two approaches.